### DISCUSSION TITLE
[ERROR] UMAT not found in:

### DISCUSSION THREAD

**Discussion by kyeongsunkim** [Anura3D/Anura3D_OpenSource#38] [Q&A] [answered]
Hi researchers, 
I am PhD student in Geotechnical Engineering Lab. in SNU.

Trying to run elemental triaxial test with external soil model, I encountered an error message "UMAT not found in: ..." (picture below):

![image](https://user-images.githubusercontent.com/88410494/198547655-39a43b54-2cc0-41d8-9679-5337be76672f.png)

In GiD v14, I set the Materials parameters as:

![image](https://user-images.githubusercontent.com/88410494/198548140-6e4e17b2-e350-4bff-b9e3-703e7d96b5d6.png)

I put EMC.dll file, which I obtained from compiling Visual STudio V2017, into the folder (url: .../problemtypes/Anura3D_2022.gid/dll):

![image](https://user-images.githubusercontent.com/88410494/198548443-45946ce9-2460-431f-b18c-7a4a25853b09.png)

Same EMC.dll file is in the folder where calculate batch file is located:

![image](https://user-images.githubusercontent.com/88410494/198548681-bb275e9a-c633-4d8f-98e2-4504dd767003.png)

What could be the possible cause of this error? Please help.
Thank you in advance.

Kyeong-Sun Kim

**Comment by murphy-john** [comment, marked as answer]
Hello @kyeongsunkim, Have you tried debugging the code in Visual Studio? You can search for the error code and put a breakpoint there. Then run the code in debug and look line by line. 

I don't have much experience with the ESMs or UMATs. It appears the code is able to find your dll, but the error gets thrown because it cannot find the UMAT part within the dll. Debugging the code should lead you to why this happens.

**Reply by WaveHello** [reply]
@kyeongsunkim another option is to hardcode the UMAT into Anura3D.  Here's a post with some information and links to go about doing that https://github.com/Anura3D/Anura3D_OpenSource/discussions/181
